src/main/java/reposense/git/GitLsTree.java

```diff
@@ -0,0 +1,81 @@
+package reposense.git;
+
+import static reposense.system.CommandRunner.runCommand;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import reposense.git.exception.GitCloneException;
+import reposense.git.exception.InvalidFilePathException;
+import reposense.model.RepoConfiguration;
+import reposense.system.LogsManager;
+import reposense.util.StringsUtil;
+import reposense.util.SystemUtil;
+
+/**
+ * Contains git ls tree related functionalities.
+ * Git ls tree is responsible to obtain the list of staged files of a branch.
+ */
+public class GitLsTree {
+
+    private static final Logger logger = LogsManager.getLogger(GitLsTree.class);
+    private static final String MESSAGE_INVALID_PATH = "Invalid filepath: %s contains %s";
+
+    // Although forward-slash (/) is an invalid character in Windows file path, it is not used as the forward-slash in
+    // the output of git-ls-tree is ambiguous. i.e. /path/to/test\file or /path/contains:a/file.
+    // Also, it is not possible to create and commit such a file on Unix systems.
+    private static final Pattern ILLEGAL_WINDOWS_CHARACTER_PATTERN = Pattern.compile("[:\\\\*?|<>:\"]");
+
+
+    /**
+     * Checks that the repository in {@code config} contains invalid file paths that are not compatible with Windows.
```

Comment:
`Verifies that the repository in {@code config} contains only file paths that are compatible with Windows.`